spring-kafka/src/main/java/org/springframework/kafka/core/KStreamBuilderFactoryBean.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.core;
+
+import java.util.Map;
+
+import org.apache.kafka.streams.KafkaClientSupplier;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.kstream.KStreamBuilder;
+import org.apache.kafka.streams.processor.internals.DefaultKafkaClientSupplier;
+
+import org.springframework.beans.factory.config.AbstractFactoryBean;
+import org.springframework.context.SmartLifecycle;
+import org.springframework.kafka.KafkaException;
+import org.springframework.util.Assert;
+
+/**
+ * An {@link AbstractFactoryBean} for the {@link KStreamBuilder} instance
+ * and lifecycle control for the internal {@link KafkaStreams} instance.
+ *
+ * @author Artem Bilan
+ *
+ * @since 2.0
+ */
+public class KStreamBuilderFactoryBean extends AbstractFactoryBean<KStreamBuilder> implements SmartLifecycle {
```

Comment:
After some private discussion decided do not follow with extension practice because that may lead to unexpected breaking changes from the target Apache Kafka API.